transport/grpc/outbound.go

```diff
@@ -131,6 +134,21 @@ func (o *Outbound) Call(ctx context.Context, request *transport.Request) (*trans
 	}, invokeErr
 }
 
+func validateRequest(req *transport.Request) error {
+	for _, v := range req.Headers.Items() {
```

Comment:
In the comment below, should we call out that this should be done by grpc-go, and link to any existing tasks on grpc-go that track fixing it there?

if grpc-go starts validating this and rejecting requests, would we still need to validate ourselves or would we remove it?

Reply:
> if grpc-go starts validating this and rejecting requests, would we still need to validate ourselves or would we remove it?
No we don't need to if grpc-go fixes it. I will add a comment like `Once XXX is resolved, we don't need this validation on the request header values.`